### PR TITLE
Update Reference and ProjectReference ordering in our targets.

### PR DESCRIPTION
**Pull request by weshaggard**
Switch the to using a common DefaultReference notion between targets
and eliminated targetingpack.props. Moved the default references for
tests into tests.targets.

Fixed where we added Private to References because in some cases it wasn't happing
because the target that set Private=true was run before all the references were
added. This is why we are now setting Reference Private=false in a different target
from ProjectReference Private=false because they sometimes need to happen at different
times in the target execution.

Set _FindDependencies=false to block RAR from trying to automatically
add the closure of the primary references as it added automatic references that folks
could accidently start depending on and we want to be explicit about our dependencies.
This was really noticable once we started auto referencing the netstandard facade which
caused everything to get pulled into the reference set.

To fix the various build scenarios where we add ProjectReferences automatically we needed
a common target to hook on which we now have AddProjectReferencesDynamically target that
correctly hooks into the various dependency chains. See comment in targets file for more
details.

cc @ericstj @mellinoe @joperezr 

Fixes https://github.com/dotnet/corefx/issues/15847, https://github.com/dotnet/corefx/issues/15744

### CODE REVIEW

**Comment by weshaggard** [comment]
merging but please do leave any feedback and I will address.

File: src/System.Linq.Expressions/tests/TestExtensions/TestOrderer.cs [hunk 1]
```diff
@@ -16,7 +16,7 @@ IEnumerable<TTestCase> ITestCaseOrderer.OrderTestCases<TTestCase>(IEnumerable<TT
             Dictionary<int, List<TTestCase>> queue = new Dictionary<int, List<TTestCase>>();
             foreach (TTestCase testCase in testCases)
             {
-                Xunit.Abstractions.IAttributeInfo orderAttribute = ReflectionAbstractionExtensions.GetCustomAttributes(testCase.TestMethod.Method, typeof(TestOrderAttribute)).FirstOrDefault();
+                Xunit.Abstractions.IAttributeInfo orderAttribute = testCase.TestMethod.Method.GetCustomAttributes(typeof(TestOrderAttribute)).FirstOrDefault();
```
Comment (mellinoe) [inline src/System.Linq.Expressions/tests/TestExtensions/TestOrderer.cs]: This seems unusual, why was this necessary?

**Comment by mellinoe** [comment]
LGTM

File: src/System.Linq.Expressions/tests/TestExtensions/TestOrderer.cs (hunk 1 above)
Comment (weshaggard) [inline src/System.Linq.Expressions/tests/TestExtensions/TestOrderer.cs]: After I changes some of default references this failed to compile and I looked at the history of the file and for some reason when you ported this originally you changed it from the extension method call to a static call so I changed it back here. I also decided to directly add the necessary reference to the xunit.execution directly to this project instead of as part of the default references.

Comment (mellinoe) [inline src/System.Linq.Expressions/tests/TestExtensions/TestOrderer.cs]: Hm, strange. I believe I changed it because having the call this way was ambiguous (resulting in compiler errors) at the time, but if that is no longer a problem then this way is good.
